tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.m

```diff
@@ -0,0 +1,269 @@
+// limitations under the License.
+
+#import "tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecord.h"
+
+#import "tensorflow_lite_support/ios/sources/TFLCommonUtils.h"
+#import "tensorflow_lite_support/ios/task/audio/core/audio_record/sources/TFLAudioRecordErrorCode.h"
+#import "tensorflow_lite_support/ios/task/audio/core/sources/TFLRingBuffer.h"
+
+#import <AVFoundation/AVFoundation.h>
+
+#define SUPPORTED_CHANNEL_COUNT 2
+
+/** Error domain for TFL Audio Record Errors. */
+static NSString *const TFLAudioRecordErrorDomain = @"org.tensorflow.lite.audio.record";
+
+@implementation TFLAudioRecord {
+  AVAudioEngine *_audioEngine;
+
+  /* Specifying a custom buffer size on AVAUdioEngine while tapping does not take effect. Hence we
+   * are storing the returned samples in a ring buffer to acheive the desired buffer size. If
+   * specified buffer size is shorter than the buffer size supported by `AVAUdioEngine` only the most
+   * recent data of the buffer of size `bufferSize` will be stored by the ring buffer. */
+  TFLRingBuffer *_ringBuffer;
+  dispatch_queue_t _conversionQueue;
+  NSError *_globalError;
+}
+
+- (nullable instancetype)initWithAudioFormat:(TFLAudioFormat *)audioFormat
+                                  bufferSize:(NSUInteger)bufferSize
+                                       error:(NSError *_Nullable *)error {
+  self = [self init];
+  if (self) {
+    if (audioFormat.channelCount > SUPPORTED_CHANNEL_COUNT) {
+      [TFLCommonUtils
+          createCustomError:error
+                 withDomain:TFLAudioRecordErrorDomain
+                       code:TFLAudioRecordErrorCodeInvalidArgumentError
+                description:@"The channel count provided does not match the supported "
+                            @"channel count. Only upto 2 audio channels are currently supported."];
+      return nil;
+    }
+
+    if (bufferSize % audioFormat.channelCount != 0) {
+      [TFLCommonUtils
+          createCustomError:error
+                 withDomain:TFLAudioRecordErrorDomain
+                       code:TFLAudioRecordErrorCodeInvalidArgumentError
+                description:@"The buffer size provided is not a multiple of channel count."];
+      return nil;
+    }
+
+    NSError *waitError = nil;
+    [TFLCommonUtils
+        createCustomError:&waitError
+               withDomain:TFLAudioRecordErrorDomain
+                     code:TFLAudioRecordErrorCodeWaitingForNewMicInputError
+              description:@"TFLAudioRecord hasn't started receiving samples from the audio "
+                          @"input source. Please wait for the input."];
+
+    _globalError = waitError;
+    _audioFormat = audioFormat;
+    _audioEngine = [[AVAudioEngine alloc] init];
+    _bufferSize = bufferSize;
+
+    _ringBuffer = [[TFLRingBuffer alloc] initWithBufferSize:_bufferSize];
+    _conversionQueue =
+        dispatch_queue_create("com.tflAudio.AudioConversionQueue", NULL);  // Serial Queue
```

Comment:
Please use `org.tensorflow.lite.AudioConversionQueue` to correctly align with the TFLite namespace.